src/services/codefixes/fixUnusedLabel.ts

```diff
@@ -0,0 +1,20 @@
+/* @internal */
+namespace ts.codefix {
+    const fixId = "fixUnusedLabel";
+    const errorCodes = [Diagnostics.Unused_label.code];
+    registerCodeFix({
+        errorCodes,
+        getCodeActions(context) {
+            const changes = textChanges.ChangeTracker.with(context, t => doChange(t, context.sourceFile, context.span.start));
+            return [createCodeFixAction(fixId, changes, Diagnostics.Remove_unused_label, fixId, Diagnostics.Remove_all_unused_labels)];
+        },
+        fixIds: [fixId],
+        getAllCodeActions: context => codeFixAll(context, errorCodes, (changes, diag) => doChange(changes, diag.file, diag.start)),
+    });
+
+    function doChange(changes: textChanges.ChangeTracker, sourceFile: SourceFile, start: number): void {
+        const token = getTokenAtPosition(sourceFile, start, /*includeJsDocComment*/ false);
+        const statement = cast(token.parent, isLabeledStatement).statement;
+        changes.deleteRange(sourceFile, { pos: token.getStart(sourceFile), end: statement.getStart(sourceFile) });

```

Comment:
this will remove comments on the statement before the label.. if you are deleting a range, why not just delete from token.pos to `:`.end?

Reply:
`/*not deleted*/ label /*deleted*/ : /*deleted*/ call();`?